### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by Mzuha**

### CODE REVIEW

File: src/main/java/mate/academy/Main.java [hunk 1]
```diff
@@ -6,15 +6,19 @@
...
+    private static final Logger logger = LogManager.getLogger(Main.class);
+
     public static void main(String[] args) {
         AuthenticationService authenticationService = new AuthenticationServiceImpl();
         User user;
         try {
             user = authenticationService.login("bob", "1234");
         } catch (AuthenticationException e) {
-            e.printStackTrace();
+            logger.error("Cant authenticate user with login = bob");
```
Comment (nikdagaev) [inline src/main/java/mate/academy/Main.java:21]: I think it's better to log error info here insted of hardcoding login = b, we may have different user here
logger.error("Cant authenticate user ", e);

File: src/main/java/mate/academy/service/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -2,11 +2,15 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Logger logger = LogManager.getLogger(AuthenticationServiceImpl.class);
+
     @Override
     public User login(String login, String password) throws AuthenticationException {
-        //TODO: add corresponding log message about method login was called
+        logger.debug("login method was called");
```
Comment (nikdagaev) [inline src/main/java/mate/academy/service/AuthenticationServiceImpl.java:13]: We may also add user info in the log message here
logger.debug("login method was called . Parameters: login: {}, login);

File: src/main/java/mate/academy/service/OrderServiceImpl.java [hunk 3]
```diff
@@ -4,27 +4,27 @@
...
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 public class OrderServiceImpl implements OrderService {
+    private static final Logger logger = LogManager.getLogger(OrderServiceImpl.class);
+
     @Override
     public Order completeOrder(Long userId) {
-        // TODO: add log message about method completeOrder was called
+        logger.debug("completeOrder method was called");
```
Comment (nikdagaev) [inline src/main/java/mate/academy/service/OrderServiceImpl.java:15]: same here

**Review by nikdagaev — CHANGES_REQUESTED** [review]

**Review by nikdagaev — APPROVED** [review]

File: src/main/java/mate/academy/resources/log4j2.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        <Logger name="mate.academy" level="debug" additivity="false">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Logger>
+        <Root level="error">
+            <AppenderRef ref="LogToFile"/>
+            <AppenderRef ref="LogToConsole"/>
+        </Root>
+    </Loggers>
+</Configuration>
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/resources/log4j2.xml:21]: add an empty line to the end of the file

File: src/main/java/mate/academy/resources/log4j2.xml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/resources/log4j2.xml:1]: does it work? can you share the screenshots of the file and the console with logs?

File: src/main/java/mate/academy/service/OrderServiceImpl.java [hunk 6]
```diff
@@ -4,27 +4,27 @@
...
 
     private List<Product> getAllProductsFromShoppingCart(Long userId) {
-        // NOTE: In production ready code this method should fetch data from DB
-        // For test purpose we simplify this method and return dummy data
         Product iphone = new Product("iPhone X", BigDecimal.valueOf(1199));
         Product macBook = new Product("MacBook Air 2020", BigDecimal.valueOf(1399));
         Product xiaomi = new Product("Xiaomi 12", BigDecimal.valueOf(499));
         List<Product> products = List.of(iphone, macBook, xiaomi);
-        // TODO: add log message about successful fetched data from DB
+        logger.debug("data was successfully fetched");
```
Comment (KseniiaMakarova) [inline src/main/java/mate/academy/service/OrderServiceImpl.java:27]: this is the message that is related to our app's business logic - and not debugging
which level should we use?

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**